### PR TITLE
add version command; also print config file when start daemon

### PR DESCRIPTION
**Pull request by merlinran**
Validating version is something I kept wanting to when chatting with miners, and even when trying myself.

Printing config file on startup can also avoid some confusions by miners.

### CODE REVIEW

File: lib/peerflags/peerflags.go [hunk 1]
```diff
@@ -180,8 +180,3 @@ func WriteConfig(v *viper.Viper, repoPathEnv, defaultRepoPath string) (string, e
 	}
 	return cf, nil
 }
-
-// MarshalConfig marshals a *viper.Viper config to JSON.
-func MarshalConfig(v *viper.Viper, pretty bool) ([]byte, error) {
-	return common.MarshalConfig(v, pretty, "private-key", "wallet-addr-sig")
```
Comment (merlinran) [inline lib/peerflags/peerflags.go:186]: drive-by. remove this and use `common.MarshalConfig` directly.

File: main.go [hunk 2]
```diff
@@ -273,9 +274,9 @@ var daemonCmd = &cobra.Command{
 		pconfig, err := peerflags.GetConfig(v, "BIDBOT_PATH", defaultConfigPath, false)
 		common.CheckErrf("getting peer config: %v", err)
 
-		settings, err := peerflags.MarshalConfig(v, !v.GetBool("log-json"))
+		settings, err := common.MarshalConfig(v, !v.GetBool("log-json"), "private-key", "wallet-addr-sig", "lotus-miner-api-token")
```
Comment (merlinran) [inline main.go:277]: added `lotus-miner-api-token` to be masked.

File: main.go [hunk 3]
```diff
@@ -380,6 +381,40 @@ var idCmd = &cobra.Command{
 	},
 }
 
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "shows current version of bidbot",
+	Args:  cobra.ExactArgs(0),
+	Run: func(c *cobra.Command, args []string) {
+		local := buildinfo.Summary()
```
Comment (merlinran) [inline main.go:389]: It's longer than expected because I want to:
1. print local version even if there's problem getting remote version.
2. if local and remote version matches, only print once.
3. print both if they don't match.

Usually local and remote and the same, but just in case if someone installs it on their laptop pointing to a deamon running on the mining server...

File: main.go [hunk 4]
```diff
@@ -380,6 +382,40 @@ var idCmd = &cobra.Command{
...
+		if res.StatusCode != http.StatusOK {
+			fail("%s: %s", res.Status, string(b))
+		}
+		remote := string(b)
+		if remote == local {
+			fmt.Println(local)
+			return
+		}
+		fmt.Printf("local%s\n", local)
+		fmt.Printf("\ndaemon%s\n", remote)
```
Comment (jsign) [inline main.go:415]: I'd add some:
`fmt.Println("WARNING! You local and remote version don't match:")`

So shouting a bit more to prevent a situation in which they forgot to restart `bidbot daemon`.

**Review by jsign — APPROVED** [review]
Nice!

File: httpapi/httpapi.go [hunk 5]
```diff
@@ -91,6 +93,10 @@ func idHandler(service Service) http.HandlerFunc {
 	}
 }
 
+func versionHandler(w http.ResponseWriter, _ *http.Request) {
+	_, _ := w.Write([]byte(buildinfo.Summary()))
```
Comment (jsign) [inline httpapi/httpapi.go:97]: As test say: `s/:=/=`.
